dot/core/sync.go

```diff
@@ -257,7 +263,33 @@ func (s *Syncer) handleBlockResponse(msg *network.BlockResponseMessage) (int64,
 				Body:   body,
 			}
 
-			// TODO: execute block and verify authorship right
+			// prepare block for sending to core_executeBlock,
+			//  core_executeBlock fails if Digest and Body data are sent
```

Comment:
it seems strange that it fails if the Body is sent, it needs the extrinsics to validate the block?

Reply:
I agree, should I try to determine why it doesn't work with extrinsics before merging this?

Reply:
that's probably a good idea, what extrinsics are you using to test? because some of them (eg. transfer) won't work because the account has no balance

Reply:
I think I'm probably not using extrinsic correctly, I was using the array at line 124 in build_test, but I'm not using in correctly.  Do you have an example of an extrinsic I can try?

Reply:
I found my issue, I should have been passing encoded `types.BlockData` into execute block (I was using  `types.Block`).